paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -107,9 +130,74 @@ def paasta_mark_for_deployment(args):
     if service and service.startswith('services-'):
         service = service.split('services-', 1)[1]
     validate_service_name(service, soa_dir=args.soa_dir)
-    return mark_for_deployment(
+    ret = mark_for_deployment(
         git_url=args.git_url,
         deploy_group=deploy_group,
         service=service,
         commit=args.commit,
     )
+    if args.block:
+        try:
+            _log(
+                service=service,
+                line="Waiting for deployment of {0} to {1} complete".format(args.commit, deploy_group),
+                component='deploy',
+                level='event'
+            )
+            wait_for_deployment(args.service, args.deploy_group,
```

Comment:
Hmm. I think we should discus this one in a standup. To me it sounds like deploy groups actually already solve this (why would you want to separate)? For the yelp main example you could have:
```
- step: prod.main
  wait_for_deployment: True
- step: some_sort_of_notification
  command: make send_notification
- step: prod.everything_else
  wait_for_deployment: False
```